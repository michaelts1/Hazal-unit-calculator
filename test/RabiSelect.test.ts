import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import RabiSelect from '../src/components/RabiSelect.vue'

describe('Component RabiSelect', () => {
	it('exists', () => {
		expect(RabiSelect).toBeTruthy()
	})

	const wrapper = mount(RabiSelect, {
		props: {
			defaultRuler: 'גר"ח',
		},
	})

	it('has a select field with correct id and 2 option elements', () => {
		expect(wrapper.find('select').exists()).toBe(true)
		expect(wrapper.find('select').attributes()).toHaveProperty('id', 'ruler-select')
		expect(wrapper.findAll('option')).toHaveLength(2)
	})

	it('matches snapshot', () => {
		expect(wrapper.html()).toMatchSnapshot()
	})

	it('reacts to value changes', async () => {
		const options = wrapper.findAll('option')

		await options[1].setSelected()
		expect(wrapper.vm.value).toBe('חזון_איש')

		await options[0].setSelected()
		expect(wrapper.vm.value).toBe('גרח')

		const rulerChangeEvents = wrapper.emitted('ruler-change') as unknown[]
		expect(rulerChangeEvents[0]).toEqual([ 'חזון_איש' ])
		expect(rulerChangeEvents[1]).toEqual([ 'גרח' ])

		const changeEvents = wrapper.emitted('change') as unknown[]
		expect(changeEvents).toHaveLength(2)
	})
})
