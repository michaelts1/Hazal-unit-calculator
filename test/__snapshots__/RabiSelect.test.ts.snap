// Vitest Snapshot v1

exports[`Component SelectRabi > matches snapshot 1`] = `
"<label for=\\"ruler-select\\" data-v-2eca11fc=\\"\\">שיטת מדידה:</label><select id=\\"ruler-select\\" class=\\"form-select\\" data-v-2eca11fc=\\"\\">
  <option value=\\"גרח\\" data-v-2eca11fc=\\"\\"> גר\\"ח (רב חיים נאה) </option>
  <option value=\\"חזון_איש\\" data-v-2eca11fc=\\"\\"> חזון איש </option>
</select>"
`;
